Stripe/STPAnalyticsClient.m

```diff
@@ -0,0 +1,102 @@
+//
+//  STPAnalyticsClient.m
+//  Stripe
+//
+//  Created by Ben Guo on 4/22/16.
+//  Copyright © 2016 Stripe, Inc. All rights reserved.
+//
+
+#import "STPAnalyticsClient.h"
+#import "STPFormEncoder.h"
+#import "NSMutableURLRequest+Stripe.h"
+#import "STPAPIClient.h"
+#if TARGET_OS_IPHONE
+#import <UIKit/UIKit.h>
+#import <sys/utsname.h>
+#endif
+
+@interface Stripe (Internal_Analytics)
++ (BOOL)shouldCollectAnalytics;
+@end
+
+@interface STPAPIClient (Internal_Analytics)
++ (NSString *)stripeUserAgentDetails;
+@end
+
+@interface STPAnalyticsClient()
+
+@property (nonatomic, readwrite) NSURLSession *urlSession;
+@property (nonatomic, readwrite) NSURL *baseURL;
+
+@end
+
+@implementation STPAnalyticsClient
+
++ (NSString *)stripeUserAgentDetails {
+    NSMutableDictionary *details = [@{
+        @"bindings_version": STPSDKVersion,
+    } mutableCopy];
+#if TARGET_OS_IPHONE
+    NSString *version = [UIDevice currentDevice].systemVersion;
+    if (version) {
+        details[@"os_version"] = version;
+    }
+    struct utsname systemInfo;
+    uname(&systemInfo);
+    NSString *deviceType = @(systemInfo.machine);
+    if (deviceType) {
+        details[@"type"] = deviceType;
+    }
+#endif
+    return [[NSString alloc] initWithData:[NSJSONSerialization dataWithJSONObject:[details copy] options:0 error:NULL] encoding:NSUTF8StringEncoding];
+}
+
++ (BOOL)shouldCollectAnalytics {
+    return NSClassFromString(@"XCTest") == nil && [Stripe shouldCollectAnalytics];
+}
+
++ (NSNumber *)timestampWithDate:(NSDate *)date {
+    return @((NSInteger)([date timeIntervalSince1970]*1000));
+}
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        NSURLSessionConfiguration *config = [NSURLSessionConfiguration defaultSessionConfiguration];
+        config.HTTPAdditionalHeaders = @{
```

Comment:
I don't know if it makes sense to still lurk this into a "user agent that isn't actually a user agent" header - can you just put these in the payload instead?